src/NuGet.Core/NuGet.Common/Telemetry/TelemetryActivity.cs

```diff
@@ -108,6 +108,7 @@ public void Dispose()
                 }
 
                 NuGetTelemetryService.EmitTelemetryEvent(TelemetryEvent);
```

Comment:
It might be easier to overload the telemetry service abstraction instead. 

The idea behind the implementation currently is that the "owner" tool actually sets up this metadata collection.

Reply:
@nkolev92 are you proposing that nuget.exe, msbuild restore and static graph restore entry points all register a "telemetry reporter", similar to how the VS entry point does, but this one just reports ETW events? Sounds like a good idea to me.

Or are you proposing something else?

Reply:
That's exactly what I'm proposing. 

The telemetry reporting or lack thereof is a product decision and it's already modeled as such.